docs/development/plugins-job-types.md

```diff
@@ -1,482 +1,372 @@
-and the dockerfile template path we created earlier.
+<details>
+  <summary>File `go-job-type/plugin.py`</summary>
 
 ```python
-from typing import Dict, Tuple
+from __future__ import annotations
 from pathlib import Path
-
-
 class Plugin:
-    def job_types(self) -> dict[str, list[tuple[Path, Path]]]:
+    def job_types(self) -> dict[str, tuple[Path, Path]]:
         """
         Job types provided by this plugin
-        :return dict of job type name (with version) -> list of images: (base image path, dockerfile template path)
+        :return dict of job type name (with version) -> (base image path, dockerfile template path)
         """
         return {
-            f'rust:{self.plugin_manifest.version}': [(
+            f'golang:{self.plugin_manifest.version}': (
                 self.plugin_dir / 'base.Dockerfile',
                 self.plugin_dir / 'job-template.Dockerfile',
-            )],
+            ),
         }
 ```
+</details>
 
-### 7. Create `.racetrackignore` file
-We don't need to incorporate all the local files into a plugin ZIP file.
-We can instruct `racetrack` plugin bundler to ignore these files by adding `.racetrackignore` file:
+Finally, also create the `.racetrackignore` file:
 ```
-Cargo.lock
-target
-.gitignore
-
+Makefile
+go.sum
 ```
 
 ### 8. Bundle plugin into a ZIP file
-Local source code of the plugin can be turned into a ZIP file
-by means of a `racetrack` client tool.
-Install it with: 
+Install the racetrack client
 ```shell
 python3 -m pip install --upgrade racetrack-client
 ```
 
-Let's run `racetrack plugin bundle` in a directory where the plugin is located (`rust-job-type` dir)
-to turn a plugin into a ZIP file.
-The outcome is `rust-job-type-1.0.0.zip`.
-
-## Installing plugin to Racetrack
-
-Now, we can make use of the plugin by installing it to Racetrack.
-
-Let's go to the Dashboard Administration page
-(you need to be staff user to see this tab)
-and upload the zipped plugin there.
-
-## Deploying sample Job
-
-Let's create an exemplary Job `sample-rust-function` that will be deployed to Racetrack.
-
-`mod.rs` file contains the logic we want to deploy:
-```rust
-use std::collections::HashMap;
-use serde_json::{Value, json};
-
-pub fn perform(input: HashMap<String, Value>) -> Value {
-    let numbers = input.get("numbers").unwrap();
-    let numbers: &Vec<Value> = numbers.as_array().unwrap();
-
-    // Sum given numbers
-    let sum = numbers.iter().fold(0, |acc, x| acc + x.as_i64().unwrap());
-
-    json!(sum)
-}
-```
-
-and the Job manifest `job.yaml` might look like this:
-```yaml
-name: rust-function
-owner_email: sample@example.com
-lang: rust
-
-git:
-  remote: https://github.com/TheRacetrack/plugin-rust-job-type
-  directory: sample-rust-function
-```
-
-Don't forget to push the Job code to a git repository.
-
-Finally, we can deploy this Job to Racetrack:
-```bash
-racetrack deploy
-```
+And then run `racetrack plugin bundle` in the `go-job-type` directory.
```

Comment:
How about adding a "Frequently Asked Questions" section at the bottom?
It may explain a difference between base image and job template, which in my opinion can be a source of a common confusion.


## Frequently Asked Questions
### What distinguishes a base image from a job template?

The job image creation process involves two stages:

1. **Preparing the base image** - 
  The base image comprises files that are common to all jobs, such as your wrapper code, and isn't tied to any specific job or manifest.

2. **Building a job from the template** - 
  Before a job is deployed, the job template is combined with a user-submitted manifest to create the final Dockerfile that will be used to build the job image.
  The job image is an extension of the base image, with the job source code added during the building process. This stage is unique to each job.

Reply:
I just created an FAQ section, is it okay or would you rather I just took what you've written here?

Reply:
It's alright.